src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -5043,6 +5043,228 @@ class C<T>
             });
         }
 
+        [Fact]
+        public void TestMainAttributes()
+        {
+            var source = CreateCompilation(@"
+using System;
+
+[main: My(""one"")]
+Console.WriteLine(""Hello World"");
+
+public class MyAttribute : Attribute { public MyAttribute(string name) {} }
+");
+
+            var mainAttrs = source.GetEntryPoint(default).GetAttributes();
+
+            var attribute = Assert.Single(mainAttrs);
+            attribute.VerifyValue(0, TypedConstantKind.Primitive, "one");
+            Assert.Equal(@"MyAttribute(""one"")", attribute.ToString());
+
+            CompileAndVerify(source, expectedOutput: "Hello World");
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void TestMainAttributes_IL()
```

Comment:
>TestMainAttributes_IL [](http://example.com/codeflow?start=20&length=21)

It feels like this unit test can be merged with the previous one, there is really no much benefit in splitting source and metadata validation.